codenvy-api-factory/src/main/java/com/codenvy/api/factory/FactoryService.java

```diff
@@ -350,6 +350,78 @@ public void removeFactory(@ApiParam(value = "Factory ID", required = true)
         factoryStore.removeFactory(id);
     }
 
+
+    /**
+     * Updates factory with a new factory content
+     *
+     * @param id
+     *         id of factory
+     * @param newFactory
+     *         the new data for the factory
+     * @throws NotFoundException
+     *         when factory with given id doesn't exist
+     */
+    @ApiOperation(value = "Updates factory information by its ID",
+            notes = "Updates factory based on the Factory ID which is passed in a path parameter")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 403, message = "User not authorized to call this operation"),
+            @ApiResponse(code = 404, message = "Factory not found"),
+            @ApiResponse(code = 500, message = "Internal Server Error")})
+    @POST
+    @Path("/{id}")
+    @RolesAllowed("user")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Factory updateFactory(@ApiParam(value = "Factory ID", required = true)
+                                 @PathParam("id") String id,
+                                 Factory newFactory) throws ForbiddenException, NotFoundException, ApiException {
+
+        // Check we've a user
+        final User user = EnvironmentContext.getCurrent().getUser();
+        if (user == null) {
+            // well this shouldn't happen if only user is authorized to call the method
```

Comment:
Sorry for the misunderstanding. I meant that user can be null if he is not authenticated. If method is restricted with @RolesAllowed("user")  you don't need that check